tests/test_private_vlan.rb

```diff
@@ -0,0 +1,777 @@
+# NXAPI New test for feature private-vlan
+# Davide Celotto Febraury 2016
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/vlan'
+
+include Cisco
+
+# TestVlan - Minitest for Vlan node utility
+class TestVlan < CiscoTestCase
+  @skip_unless_supported = 'vlan'
+
+  @@cleaned = false # rubocop:disable Style/ClassVars
+  def cleanup
+    Vlan.vlans.each do |vlan, obj|
+      # skip reserved vlans
+      next if vlan == '1'
+      next if node.product_id[/N5K|N6K|N7K/] && (1002..1005).include?(vlan.to_i)
+      obj.destroy
+    end
+    config('no feature vtp')
+    config('no feature private-vlan')
+  end
+
+  def setup
+    super
+    cleanup unless @@cleaned
+    @@cleaned = true # rubocop:disable Style/ClassVars
+  end
+
+  def teardown
+    cleanup
+    config('no feature private-vlan')
+    config('no feature vtp')
+  end
+
+  def test_private_type_default
+    config('no feature vtp')
+    config('feature private-vlan')
+    v1 = Vlan.new(100)
+    pv_type = ''
+    if validate_property_excluded?('vlan', 'private_vlan_type')
+      assert_nil(v1.private_vlan_type)
+    else
+      assert_equal(pv_type, v1.private_vlan_type)
+    end
+    v1.destroy
+  end
+
+  def test_private_association_default
+    config('no feature vtp')
```

Comment:
I call feature private-vlan in the setter and for the default case I'm just using the getter. So this is
just a special case for the two default tests. I feel it should be ok for just these two tests.